receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -131,10 +140,17 @@ func (kr *k8sobjectsreceiver) Shutdown(context.Context) error {
 
 func (kr *k8sobjectsreceiver) start(ctx context.Context, object *K8sObjectsConfig) {
 	resource := kr.client.Resource(*object.gvr)
-	kr.setting.Logger.Info("Started collecting", zap.Any("gvr", object.gvr), zap.Any("mode", object.Mode), zap.Any("namespaces", object.Namespaces))
+	kr.setting.Logger.Info("Started collecting",
+		zap.Any("gvr", object.gvr),
+		zap.Any("mode", object.Mode),
+		zap.Any("namespaces", object.Namespaces))
 
 	switch object.Mode {
 	case PullMode:
+		// Set default interval if not set
+		if object.Interval == 0 {
```

Comment:
This does not seem the right place to do this. Could that be done earlier when we originally create/validate the objects?

Reply:
Moved it to `newReceiver`